pkg/query/storeset.go

```diff
@@ -0,0 +1,194 @@
+package query
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/store"
+	"github.com/improbable-eng/thanos/pkg/store/storepb"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+)
+
+// StoreSet maintains a set of active stores. It is backed by a peer's view of the cluster
+// and a list of static store addresses.
```

Comment:
Comment is in theory leaking information about what main choses to do to implement the `storeAddrs` func.